dvc/dependency/gdrive.py

```diff
@@ -0,0 +1,16 @@
+from dvc.dependency.base import BaseDependency
+from dvc.output import BaseOutput
+from dvc.tree import GDriveTree
+from dvc.tree.gdrive import GDriveURLInfo
+
+
+class GDriveDependency(BaseDependency, BaseOutput):
+    TREE_CLS = GDriveTree
+
+    def __init__(self, stage, path, info=None, **wkwargs):
+        repo = stage.repo if stage else None
+        if "tree" not in wkwargs:
+            base_path = GDriveURLInfo(path).replace(path="").url
+            tree = GDriveTree(repo, {"url": base_path})
+            wkwargs["tree"] = tree
```

Comment:
Could you clarify why we need this for gdrive but not for any other dep?

Reply:
In `GDriveTree`, most of the methods rely on `_get_item_id`, which depends on `self._bucket` to work properly. `BaseOutput` (the class from which `GDriveDependency` inherits) has a `tree` attribute, which, by default, is instantiated with an empty configuration:

https://github.com/iterative/dvc/blob/689a569246634ed1f8367b7a0359e334964f1252/dvc/output/base.py#L124

However, to have `self._bucket` correctly configured, we need to pass `url` in the configuration. I presume the other `Dependency` classes work as expected with a "non-configured" `tree`.

Reply:
Correct, other trees handle that as a special case: https://github.com/iterative/dvc/blob/96ead9425af0c3e3503ebb335fd3edacd2c5f6e1/dvc/tree/http.py#L43 Could we do the same for gdrive?

Reply:
If I do the same for `GDriveTree`, I'll need to change the solution of `_get_item_id` and other methods to not rely on `self._bucket`, which will demand a change at least in `_path_to_item_ids`, `_get_cached_item_ids`, `_path_to_item_ids` and `_list_params`.

Reply:
To make my comment more clear: `GDriveTree` needs `url` (and the `_bucket` from it) to work properly. If we allow a `url=None`, we need to change at least those methods I mentioned to get the `bucket` from the `info_path` passed by the param of the public methods. Such a change may also affect the cache mechanism that relies on `_bucket`.

Reply:
I see. The problem with using `self._bucket` for things like import is that you are mixing up the concepts of tree and the path that belongs to that tree. `self.path_info`(along with self._bucket) should only be used explicitly, like in s3 and other trees. If we are operating with a third-party `path_info`, we should use it or otherwise there might be a difference in tree and path configuration that we won't notice.

Reply:
I've written new code without relying on `self._bucket`.

I've noticed, however,  in the current tests,`GDriveDependency` is always getting a `tree` based on the remote tree, because the tests are based on `remote://` urls.

https://github.com/iterative/dvc/blob/7501302baa6f724a2e962446249044a563b44568/dvc/dependency/__init__.py#L65-L67

So, the case where `self._bucket` is `None` was never tested. Now I added a test to import a `gdrive://` url.

The branch was also rebased.